tasks/install_powershell.ps1

```diff
@@ -0,0 +1,50 @@
+[CmdletBinding()]
+Param(
+	[String]$version
+)
+# If an error is encountered, the script will stop instead of the default of "Continue"
+$ErrorActionPreference = "Stop"
+
+$arch = if ((Get-WmiObject Win32_OperatingSystem).OSArchitecture -match '^32') {"x86"} else {"x64"}
+
+if ($version) {
+    $msi_source = "http://downloads.puppetlabs.com/windows/puppet5/puppet-agent-${version}-${arch}.msi"
+}
+else {
+    $msi_source = "http://downloads.puppetlabs.com/windows/puppet5/puppet-agent-${arch}-latest.msi"
+}
+
+$msi_dest = Join-Path ([System.IO.Path]::GetTempPath()) "puppet-agent-$arch.msi"
+
+function DownloadPuppet {
+  Write-Verbose "Downloading the Puppet Agent for Puppet Enterprise on $env:COMPUTERNAME..."
+
+  $webclient = New-Object system.net.webclient
+
+  try {
+    $webclient.DownloadFile($msi_source,$msi_dest)
```

Comment:
Have you tested to see what happens if `$msi_dest ` already exists?

Reply:
It appears that `$msi_dest` will be over-written. I think this is the desired behavior.